test/spec/proxySpec.js

```diff
@@ -1148,6 +1146,86 @@ describe('proxy', function() {
       // }, 100);
     });
   });
+  describe('Already proxified values', function() {
+    it("shouldn't re-proxify (it should deep clone) proxified values from different instances", function() {
```

Comment:
Well, Tomek's example can be true today if it pushed a proxified object to an array. 

But I don't really understand the example. Is `reviewState` a JSONPatcherProxy instance's proxy? Or just another proxy?